File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/ok/OkResponse.java
package ru.kmorozov.gbd.core.logic.connectors.ok;

import ru.kmorozov.gbd.core.logic.connectors.Response;

import java.io.InputStream;

/**
 * Created by km on 17.05.2016.
 */
public class OkResponse implements Response {

    private final okhttp3.Response resp;

    OkResponse(okhttp3.Response resp) {
        this.resp = resp;
    }

    @Override
    public InputStream getContent() {
        return this.resp.body().byteStream();
    }

    @Override
    public String getImageFormat() {
        return this.resp.body().contentType().subtype();
    }

    @Override
    public void close() {
        this.resp.body().close();
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/ok/OkResponse.kt
package ru.kmorozov.gbd.core.logic.connectors.ok

import ru.kmorozov.gbd.core.logic.connectors.Response

import java.io.InputStream

/**
 * Created by km on 17.05.2016.
 */
class OkResponse internal constructor(private val response: okhttp3.Response) : Response {
    override val statusCode: Int
        get() = response.code()

    override val content: InputStream
        get() = response.body()!!.byteStream()

    override val imageFormat: String
        get() = response.body()!!.contentType()!!.subtype()

    override fun close() {
        response.body()!!.close()
    }
}
